app/src/org/commcare/dalvik/activities/LoginActivity.java

```diff
@@ -566,4 +586,61 @@ public CustomProgressDialog generateProgressDialog(int taskId) {
     public boolean isBackEnabled() {
         return false;
     }
+    
+    private void refreshView() {
+        Spinner spinner = (Spinner) findViewById(R.id.app_selection_spinner);
+        TextView message = (TextView) findViewById(R.id.welcome_msg);
+        ArrayList<ApplicationRecord> readyApps = CommCareApplication._().getReadyAppRecords();
+        if (readyApps.size() <= 1) {
+            spinner.setVisibility(View.GONE);
+            message.setText(R.string.login_welcome_single);
+            return;
+        }
+        message.setText(R.string.login_welcome_multiple);
+        ArrayList<String> appNames = new ArrayList<>();
+        ArrayList<String> appIds = new ArrayList<>();
+        for (ApplicationRecord r : readyApps) {
+            String name = r.getDisplayName();
+            appNames.add(name);
+            appIds.add(r.getUniqueId());
+            namesToRecords.put(name, r);
+        }
+        ArrayAdapter<String> adapter = new ArrayAdapter<>(this, R.layout.spinner_text_view, appNames);
+        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        spinner.setAdapter(adapter);
+        spinner.setOnItemSelectedListener(this);
+        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);

```

Comment:
When we get here we should already have seated an app on login, so we should just be able to query for the key of the current logged in app (which will prevent things from getting out of sync).